packages/react-button/src/components/MenuButton/useMenuButton.ts

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import { getCode, ArrowDownKey } from '@fluentui/keyboard-key';
+import { ComposePreparedOptions } from '@fluentui/react-compose';
+import { useControllableValue, useMergedRefs } from '@uifabric/react-hooks';
+import { DirectionalHint } from 'office-ui-fabric-react';
+import { useButton } from '../Button/useButton';
+import { MenuButtonProps, MenuButtonState } from './MenuButton.types';
+
+/**
+ * The useMenuButton hook processes the MenuButton component props and returns state.
+ * @param props - MenuButton props to derive state from.
+ */
+export const useMenuButton = (
+  props: MenuButtonProps,
+  ref: React.Ref<HTMLElement>,
+  options: ComposePreparedOptions,
+): MenuButtonState => {
+  const {
+    defaultExpanded = false,
+    expanded: controlledExpanded,
+    menu,
+    menuIcon,
+    onClick,
+    onKeyDown,
+    onMenuDismiss,
+    ...rest
+  } = props;
+  const [expanded, setExpanded] = useControllableValue(controlledExpanded, defaultExpanded);
+  const buttonRef = React.useRef(null);
+
+  const _onClick = React.useCallback(
+    (ev: React.MouseEvent<HTMLButtonElement>) => {
+      if (onClick) {
+        onClick(ev);
+
+        if (ev.defaultPrevented) {
+          return;
+        }
+      }
+
+      setExpanded(!expanded);
+    },
+    [expanded, onClick],
+  );
+
+  const onDismiss = React.useCallback(() => {
```

Comment:
don't need to `useCallback` for these handlers. creating functions should not be costly. memoize only when needed